### PR TITLE
docs(kubernetes): document annotation-driven details ui

### PR DESCRIPTION
**Pull request by ghost**
This PR documents the new annotation-driven UI functionality currently under review over here:

https://github.com/spinnaker/deck/pull/5197

### CODE REVIEW

File: guides/user/kubernetes-v2/annotations-ui/index.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+- account - the spinnaker account for this resource
+- apiVersion - the kubernetes apiVersion of this resource
+- cloudProvider - this will always be `kubernetes`
+- displayName - the name of the resource prepared for UI display
+- hasHealthStatus - a boolean indicating whether the instance has health status
+- healthState - the instance's health status, if available
+- id - the instance's id
+- kind - the kubernetes kind of this resource
+- manifest - the kubernetes mabnifest as JSON object
```
Comment (ezimanyi) [inline guides/user/kubernetes-v2/annotations-ui/index.md]: typo s/mabnifest/manifest

File: guides/user/kubernetes-v2/annotations-ui/index.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+- account - the spinnaker account for this resource
+- apiVersion - the kubernetes apiVersion of this resource
+- application - the spinnaker application name, if any, for this resource
+- cloudProvider - this will always be `kubernetes`
+- detail - the spinnaker detail, if any, for this resource
+- displayName - the name of the resource prepared for UI display
+- id - the security group's id
+- kind - the kubernetes kind of this resource
+- manifest - the kubernetes mabnifest as JSON object
```
Comment (ezimanyi) [inline guides/user/kubernetes-v2/annotations-ui/index.md]: s/mabnifest/manifest/

File: guides/user/kubernetes-v2/annotations-ui/index.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+
+#### Server Group Managers
+
+- account - the spinnaker account for this resource
+- apiVersion - the kubernetes apiVersion of this resource
+- cloudProvider - this will always be `kubernetes`
+- displayName - the name of the resource prepared for UI display
+- kind - the kubernetes kind of this resource
+- manifest - the kubernetes mabnifest as JSON object
```
Comment (ezimanyi) [inline guides/user/kubernetes-v2/annotations-ui/index.md]: s/mabnifest/manifest/

File: guides/user/kubernetes-v2/annotations-ui/index.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+#### Load Balancers
+
+- account - the spinnaker account for this resource
+- apiVersion - the kubernetes apiVersion of this resource
+- cloudProvider - this will always be `kubernetes`
+- detail - the spinnaker detail, if any, for this resource
+- displayName - the name of the resource prepared for UI display
+- kind - the kubernetes kind of this resource
+- manifest - the kubernetes mabnifest as JSON object
```
Comment (ezimanyi) [inline guides/user/kubernetes-v2/annotations-ui/index.md]: s/mabnifest/manifest/

**Review by ezimanyi — APPROVED** [review]
LGTM, just one typo in four places.

**Comment by ghost** [comment]
thanks!, fixed

File: guides/user/kubernetes-v2/annotations-ui/index.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+### Using Templates
+
+Template values can be included in the content of the annotation and will be replaced when
+they are rendered by Deck.  A templated value has the following appearance in an annotation:
+`{{ templateKey }}` where `templateKey` will vary depending on the available set of keys
+for the resource that is annotated.  The complete set of available keys is documented below.
+
+#### Instances
+
+- account - the spinnaker account for this resource
```
Comment (lwander) [inline guides/user/kubernetes-v2/annotations-ui/index.md:80]: Can we get details from the content of the account object? e.g. which context was used

Comment (ghost) [reply guides/user/kubernetes-v2/annotations-ui/index.md:80]: No I don't think we can currently.  It's whatever fields appear in the `IKubernetesInstance` interface in Deck.  Sounds like a good feature request though.

Comment (ghost) [reply guides/user/kubernetes-v2/annotations-ui/index.md:80]: https://github.com/spinnaker/spinnaker/issues/2671

Comment (lwander) [reply guides/user/kubernetes-v2/annotations-ui/index.md:80]: Awesome, sounds easy enough to implement.

**Review by lwander — APPROVED** [review]

**Pull request merged**